liblangutil/SemVerHandler.cpp

```diff
@@ -240,8 +241,12 @@ SemVerMatchExpression::MatchComponent SemVerMatchExpressionParser::parseMatchCom
 		component.levelsPresent++;
 		if (currentChar() == '.')
 			nextChar();
-		else
+		else if (currentChar() == char(-1)) // EOF
 			break;
+		else if (contains("|<>=^~- 0123456789"sv, currentChar()))
```

Comment:
Does this support combining components without space (`>5<7`)? Do we really want to support that?

Reply:
We do not enforce a whitespace in between the symbols as we are reusing the solidity `Scanner` for tokenization.

NB: Enforcing a  (same line) whitespace in between the version components would require columnar token information that we currently do not pass along.

Reply:
Ok, I see, so this is fine.

What I don't understand: How can `currentChar()` be a digit? Shouldn't it be cosumed by `parseVersionPart()`?

Reply:
The tokens are already pre-tokenized. `parseVersionPart()` is using `nextChar()` to consume more chars as long as it's a digit but stops when `m_pos` points to the next token in the the array of tokens. So the next call to `currentChar()` outside `parseVersionPart()` already points to the first char of the next token.

It's weird, I know. :)

Reply:
So you are saying this might be the case for `>=0.1 1`, right? Can you check if we have a test for that and that it parses as ">=0.1.0 && =1.*.*" instead of `>=0.11`?

Reply:
On the other hand, do we really want to list all the allowed characters here? What if we just break in all cases (and not throw SemVerError)? Will we get an error later?

Reply:
I mean does it make sense to check for EOF one level higher in the parser?